dvc/repo/plots/__init__.py

```diff
@@ -260,6 +302,168 @@ def _collect_plots(
     return result
 
 
+def _get_data_targets(pathlib, definitions: Dict):
+    result: Set = set()
+    if "data" in definitions:
+        for file_relpath, content in definitions["data"].items():
+            if "data" in content:
+                for plot_id, config in content["data"].items():
+                    result = result.union(
+                        infer_data_sources(
+                            pathlib, file_relpath, plot_id, config
+                        )
+                    )
+    return result
+
+
+def infer_data_sources(pathlib, config_path, plot_id, config):
+    import posixpath
+
+    def _deduplicate(lst: List):
+        return list({elem: None for elem in lst}.keys())
+
+    y = config.get("y", None)
+    if isinstance(y, dict):
+        sources = list(y.keys())
+    else:
+        sources = [plot_id]
+
+    # TODO bug in dvc_objects? normpath normalizes to windows path on windows
+    return _deduplicate(
+        posixpath.normpath(pathlib.join(pathlib.dirname(config_path), source))
+        for source in sources
+    )
+
+
+def _matches(targets, config_file, plot_id):
+    import re
+
+    from dvc.utils.plots import get_plot_id
+
+    if not targets:
+        return True
+
+    full_id = get_plot_id(plot_id, config_file)
+    if any(
+        (re.match(target, plot_id) or re.match(target, full_id))
+        for target in targets
+    ):
+        return True
+    return False
+
+
+def _dvcfile_relpath(repo, dvcfile):
+    fs = repo.dvcfs
+
+    # TODO from_os_path changes abs to relative
+    return fs.path.relpath(
+        fs.path.join("/", fs.from_os_path(dvcfile.path)), fs.path.getcwd()
+    )
+
+
+def _collect_output_plots(
+    repo, targets, props, onerror: Optional[Callable] = None
+):
+    fs = repo.dvcfs
+    result: Dict[str, Dict] = {}
+    for plot in repo.index.plots:
+        plot_props = _plot_props(plot)
+        dvcfile = plot.stage.dvcfile
+        config_path = _dvcfile_relpath(repo, dvcfile)
+        config_dirname = os.path.dirname(config_path)
+        if _matches(targets, config_path, str(plot)):
+            unpacked = unpack_if_dir(
+                fs,
+                fs.path.join(config_dirname, plot.def_path),
+                props={**plot_props, **props},
+                onerror=onerror,
+            )
+
+            dpath.util.merge(
+                result,
+                {"": unpacked},
+            )
+    return result
+
+
+def _collect_pipeline_files(repo, targets: List[str], props):
+    from dvc.dvcfile import PipelineFile
+
+    result: Dict[str, Dict] = {}
+    dvcfiles = {stage.dvcfile for stage in repo.index.stages}
+    for dvcfile in dvcfiles:
+        if isinstance(dvcfile, PipelineFile):
+            dvcfile_defs = dvcfile.load().get("plots", {})
+            rpath = _dvcfile_relpath(repo, dvcfile)
```

Comment:
Is there a difference if we use:

```suggestion
            rpath = dvcfile.relpath
```

Reply:
We should, but it seems like some kind of a bug. Need to investigate it further. Fix won't be included in this change. Added TODO.